quill-core/src/main/scala/io/getquill/quotation/Parsing.scala

```diff
@@ -286,6 +286,9 @@ trait Parsing {
       OptionOperation(OptionForall, astParser(o), identParser(alias), astParser(body))
     case q"$o.exists({($alias) => $body})" if (is[Option[Any]](o)) =>
       OptionOperation(OptionExists, astParser(o), identParser(alias), astParser(body))
+    case q"$o.contains[$t]($body)" if (is[Option[Any]](o)) =>
+      OptionOperation(OptionExists, astParser(o),
+        Ident("_containsAlias"), BinaryOperation(Ident("_containsAlias"), EqualityOperator.`==`, astParser(body)))
```

Comment:
I'd prefer to create a new AST for this so we don't lose information during parsing.